wp-includes/sqlite-ast/class-wp-sqlite-information-schema-reconstructor.php

```diff
@@ -0,0 +1,644 @@
+			$mysql_type = substr( $mysql_type, 0, strlen( $mysql_type ) - strlen( ' KEY' ) );
+		}
+		return $mysql_type;
+	}
+
+	/**
+	 * Get a MySQL column type from an SQLite column type.
+	 *
+	 * This method converts an SQLite column type to a MySQL column type as per
+	 * the SQLite column type affinity rules:
+	 *   https://sqlite.org/datatype3.html#determination_of_column_affinity
+	 *
+	 * @param  string $column_type The SQLite column type.
+	 * @return string              The MySQL column type.
+	 */
+	private function get_mysql_column_type( string $column_type ): string {
+		$type = strtoupper( $column_type );
+
+		/*
+		 * Following the rules of column affinity:
+		 *   https://sqlite.org/datatype3.html#determination_of_column_affinity
+		 */
+
+		// 1. If the declared type contains the string "INT" then it is assigned
+		//    INTEGER affinity.
+		if ( str_contains( $type, 'INT' ) ) {
+			return 'int';
+		}
+
+		// 2. If the declared type of the column contains any of the strings
+		//    "CHAR", "CLOB", or "TEXT" then that column has TEXT affinity.
+		if ( str_contains( $type, 'TEXT' ) || str_contains( $type, 'CHAR' ) || str_contains( $type, 'CLOB' ) ) {
+			return 'text';
+		}
+
+		// 3. If the declared type for a column contains the string "BLOB" or
+		//    if no type is specified then the column has affinity BLOB.
+		if ( str_contains( $type, 'BLOB' ) || '' === $type ) {
+			return 'blob';
+		}
+
+		// 4. If the declared type for a column contains any of the strings
+		//    "REAL", "FLOA", or "DOUB" then the column has REAL affinity.
+		if ( str_contains( $type, 'REAL' ) || str_contains( $type, 'FLOA' ) ) {
+			return 'float';
+		}
+		if ( str_contains( $type, 'DOUB' ) ) {
+			return 'double';
+		}
+
+		/**
+		 * 5. Otherwise, the affinity is NUMERIC.
+		 *
+		 * While SQLite defaults to a NUMERIC column affinity, it's better to use
+		 * TEXT in this case, because numeric SQLite columns in non-strict tables
+		 * can contain any text data as well, when it is not a well-formed number.
+		 *
+		 * See: https://sqlite.org/datatype3.html#type_affinity
+		 */
+		return 'text';
+	}
+
+	/**
+	 * Escape a string literal for MySQL DEFAULT values.
+	 *
+	 * @param  string $literal The string literal to escape.
+	 * @return string          The escaped string literal.
+	 */
+	private function escape_mysql_string_literal( string $literal ): string {
+		// See: https://www.php.net/manual/en/mysqli.real-escape-string.php
+		return "'" . addcslashes( $literal, "\0\n\r'\"\Z" ) . "'";
+	}
+
+	/**
+	 * Quote an SQLite identifier.
+	 *
+	 * Wrap the identifier in backticks and escape backtick values within.
+	 *
+	 * @param  string $unquoted_identifier The unquoted identifier value.
+	 * @return string                      The quoted identifier value.
+	 */
+	private function quote_sqlite_identifier( string $unquoted_identifier ): string {
+		return '`' . str_replace( '`', '``', $unquoted_identifier ) . '`';
```

Comment:
I think you told me once that double backticks are the only way to escape a backtick – can we mention that in the docstring with a documentation link? It just looks suspiciously easy and I don't want to mislead anyone in the future into searching for a security issue here.

Reply:
Thank you!